src/install/dracut-install.c

```diff
@@ -1779,6 +1790,11 @@ static int install_module(struct kmod_ctx *ctx, struct kmod_module *mod)
                         r = install_dependent_modules(ctx, modpost, NULL);
                         ret = ret ? : r;
                 }
+#ifdef CONFIG_WEAKDEP
+                ret = kmod_module_get_weakdeps(mod, &modweak);
```

Comment:
Assigning `ret` will overwrite failures from `install_dependent_modules`.

Reply:
Thank you for your feedback, I will fix it right now

Reply:
Retested in my local environment.